packages/light-adjustment/zig/src/main.zig

```diff
@@ -123,15 +128,23 @@ pub const Agcwd = struct {
         pdf = Pdf.fromImageAndMask(image, mask);
         const cdf = Cdf.fromPdf(&pdf.toWeightingDistribution(self.options.alpha));
         self.mapping_curve = cdf.toIntensityMappingCurve(self.options);
+        for (0..self.mapping_curve.len) |v| {
+            const nv = @intCast(u32, self.mapping_curve[v]);
+            for (0..self.rgb_table[v].len) |rgb| {
+                self.rgb_table[v][rgb] = @truncate(u8, if (v > 0) rgb * nv / v else 0);
```

Comment:
`else` 部分を `0` にしてしまうと、完全な黒は AGCWD 適用後も常に黒になってしまうので `nv` の方が適切な気がしました（ `v = 0` の時に意味があるのは `rgb = 0` だけのはずで、その場合には `v == rgb` となるので `rgb * nv / v = rgb / v * nv = (v / v = nv) = nv` になるようなイメージ）: 
```suggestion
                self.rgb_table[v][rgb] = @truncate(u8, if (v > 0) rgb * nv / v else nv);
```

Reply:
そうですね